pkg/k8s/watchers/crd.go

```diff
@@ -345,19 +348,32 @@ func newListWatchFromClient(
 // version promotion), the code still works. Hence, we do not need different
 // versions of this function to handle the different types. See
 // https://github.com/kubernetes/kubernetes/pull/77136.
-func tableToPomList(t *metav1.Table) *metav1.PartialObjectMetadataList {
-	list := &metav1.PartialObjectMetadataList{
+func tableToPomList(t *slim_metav1.Table) *slim_metav1.PartialObjectMetadataList {
+	list := &slim_metav1.PartialObjectMetadataList{
 		TypeMeta: t.TypeMeta,
 		ListMeta: t.ListMeta,
-		Items:    make([]metav1.PartialObjectMetadata, 0, len(t.Rows)),
+		Items:    make([]slim_metav1.PartialObjectMetadata, 0, len(t.Rows)),
+	}
+
+	// find column that contains the name field
+	idx := -1
+	for i, cd := range t.ColumnDefinitions {
+		if strings.ToLower(cd.Name) == "name" {
+			idx = i
+		}
+	}
+	if idx == -1 {
+		log.WithFields(logrus.Fields{
+			"column-definitions": t.ColumnDefinitions,
+		}).Error("Unable to find column definition with 'Name' field skipping")
```

Comment:
If this happens, there's an error in the initialization which might cause the Cilium initialization to hang, there's nothing that the user can do. If we change it to warning it might give the impression the user has the option to avoid this, which is not true.